### PR TITLE
Fix unassociate datasrc api of prisma store

### PR DESCRIPTION
**Pull request by S1LV3RJ1NX**

### CODE REVIEW

File: backend/modules/metadata_store/prismastore.py [hunk 1]
```diff
@@ -340,17 +340,26 @@ async def aunassociate_data_source_with_collection(
                 status_code=400,
                 detail=f"Data source with fqn {data_source_fqn} not associated with collection {collection_name}",
             )
-
         associated_data_sources.pop(data_source_fqn, None)
```
Comment (chiragjn) [inline backend/modules/metadata_store/prismastore.py:343]: It is quite confusing what is the type of associated_data_sources?

Line 324 annotates it as pydantic object, what exactly is returned from DB?

Comment (S1LV3RJ1NX) [reply backend/modules/metadata_store/prismastore.py:343]: Actually it does not it is of form dict: dict

Comment (S1LV3RJ1NX) [reply backend/modules/metadata_store/prismastore.py:341]: I have removed the type annotation. Actually from DB we get of form str: AsociatedDataSrcs

**Review by chiragjn — APPROVED** [review]
Keeping this approved, but we should check why can't we get correct data types from prisma itself

**Pull request merged**